client/src/main/java/io/pravega/client/stream/ReaderGroupConfig.java

```diff
@@ -49,11 +50,48 @@
 
     private final int maxOutstandingCheckpointRequest;
 
+    private final StreamDataRetention retentionConfig;
+
+    /**
+     * If a Stream's Retention Policy is set to {@link RetentionPolicy.RetentionType#CONSUMPTION},
+     * and the user expects reads from this Reader Group to impact data retention in the Stream,
+     * the retentionConfig in {@link ReaderGroupConfig} should be set to
+     * to 'CONSUMPTION_BASED_USER_STREAMCUT' or 'CONSUMPTION_BASED_AT_LAST_CHECKPOINT'.
+     * Setting these options implies the Reader Group will notify the Controller of its consumption position {@link StreamCut}
+     * in the Stream, so only un-consumed data can be retained.
+     * This notification can be manual ('CONSUMPTION_BASED_USER_STREAMCUT') or automatic ('CONSUMPTION_BASED_AT_LAST_CHECKPOINT')
+     * If a Stream's Retention Policy is TIME/SPACE based the ReaderGroupConfig should have retentionConfig='NO_IMPACT',
+     * so consumption position notifications are not sent to Controller.
```

Comment:
```suggestion
     * If a Stream's Retention Policy is 'TIME' or 'SPACE' based the ReaderGroupConfig will have no effect.
```

Reply:
> If a Stream's Retention Policy is 'TIME' or 'SPACE' based the ReaderGroupConfig will have no effect.

Based on discussion with @shiveshr and @fpj today, a Stream with TIME/SPACE based retention policy can be truncated based on RG Stream-Cuts. In this case the TIME/SPACE policy related paramters just serve as min/max limits for Consumption based retention.